tests/test_sed.py

```diff
@@ -1186,6 +1186,37 @@ def test_emission_line():
                 sed.atRedshift(-2.0)
 
 
+@timer
+def test_flux_type_calculateFlux():
+    sed1 = galsim.SED(
+        galsim.LookupTable([1,2,3,4,5], [1.1, 1.9, 1.4, 1.8, 2.0]),
+        wave_type='nm', flux_type='flambda'
+    )
+    sed2 = galsim.SED(
+        galsim.LookupTable([1,2,3,4,5], [1.1, 1.9, 1.4, 1.8, 2.0]),
+        wave_type='nm', flux_type='fnu'
+    )
+    sed3 = galsim.SED(
+        galsim.LookupTable([1,2,3,4,5], [1.1, 1.9, 1.4, 1.8, 2.0]),
+        wave_type='nm', flux_type='fphotons'
+    )
+    sed4 = galsim.SED(
+        galsim.LookupTable([1,2,3,4,5], [1.1, 1.9, 1.4, 1.8, 2.0]),
+        wave_type="nm", flux_type=units.Lsun / units.Hz / units.Mpc**2
+    )
+    bp = galsim.Bandpass(galsim.LookupTable([2,4], [1,2]), wave_type='nm')
+
+    for sed in [sed1, sed2, sed3, sed4]:
+        flux1 = sed.calculateFlux(bp)
+        flux2 = (1 * sed).calculateFlux(bp)
```

Comment:
Probably worth also adding a test of multiplying a gsobject by sed, rather than just 1.  Especially since that was the original way you found this bug.  I think currently that hits the same code as `1*sed`, but I can imagine them being separate code paths for some reason in the future.

Reply:
Would a flat SED be sufficient to test this? It's degenerate with multiplying by 1 but should hit `SED._mul_sed` instead of `SED._mul_scalar`